### PR TITLE
*: add some cluster metrics

### PR DESCRIPTION
**Pull request by huachaohuang**
Add some metrics to answer these questions:
- How many difference kind of balance operators?
- How many regions in the cluster?
- How many difference state of stores?
- How much data in the cluster?
- How much capacity in the cluster?
- How's the capacity balance in the cluster?
- How's the leader balance in the cluster?

### CODE REVIEW

File: server/balancer_worker.go [hunk 1]
```diff
@@ -275,3 +274,46 @@ func (bw *balancerWorker) storeScores(store *storeInfo) []int {
...
+	case balanceOP:
+		prefix = "balance"
+	}
+	for _, op := range bop.Ops {
+		if _, ok := op.(*onceOperator); ok {
+			op = op.(*onceOperator)
+		}
+		switch o := op.(type) {
+		case *changePeerOperator:
+			metrics[prefix+o.Name]++
```
Comment (siddontang) [inline server/balancer_worker.go]: need o.Name here?

Comment (huachaohuang) [inline server/balancer_worker.go]: Yep, like "balance_add_peer" or "balance_remove_peer", but I forgot to add a underline...

File: server/balancer_worker.go [hunk 2]
```diff
@@ -275,3 +274,46 @@ func (bw *balancerWorker) storeScores(store *storeInfo) []int {
 
 	return scores
 }
+
+func collectOperatorMetrics(bop *balanceOperator) map[string]uint64 {
+	metrics := make(map[string]uint64)
+	prefix := ""
+	switch bop.Type {
+	case adminOP:
+		prefix = "admin"
```
Comment (siddontang) [inline server/balancer_worker.go]: why not using "admin_" directly?

Comment (huachaohuang) [inline server/balancer_worker.go]: I think both are fine, but "admin_" looks odd to me.

Comment (siddontang) [inline server/balancer_worker.go]: can reduce a string re-malloc.

**Comment by siddontang** [comment]
LGTM

**Comment by huachaohuang** [comment]
@disksing PTAL

File: server/cluster.go [hunk 3]
```diff
@@ -508,6 +508,62 @@ func (c *RaftCluster) checkStores() {
 	}
 }
 
+func (c *RaftCluster) collectMetrics() {
+	cluster := c.cachedCluster
+	metrics := make(map[string]float64)
+	regionTotalCount := 0
+	minUsedRatio, maxUsedRatio := float64(0.0), float64(0.0)
+	minLeaderRatio, maxLeaderRatio := float64(0.0), float64(0.0)
```
Comment (disksing) [inline server/cluster.go]: I suppose `minXxRatio` should be initialized to 1.

**Comment by disksing** [comment]
LGTM

**Pull request merged**